### DISCUSSION TITLE
Information on supported Keras layers in IREE

### DISCUSSION THREAD

**Discussion by HemaSowjanyaMamidi** [iree-org/iree#9358] [Q&A] [answered]
Hi All,
@powderluv, @benvanik, @stellaraccident
The link below gives the supported layers in Keras and is updated on 2020/12/8. Can I get the latest set of Keras layers that are supported by IREE? and also I would like to contribute to some of the operators. It would be very helpful if there is any documentation or PR's that can lead me on how to implement and contribute. I would also like to understand whether an dialect implemented in MLIR can be used as an operator in IREE. Suppose 'conv' is a dialect I implemented in MLIR, can that be used in IREE?

https://github.com/google/iree/blob/main/docs/developers/tensorflow_coverage/tf_keras_coverage.md

Thanks,
Hema Sowjanya Mamidi

**Comment by stellaraccident** [comment, marked as answer]
Ah, this doc is from the non published version of our original developer docs. We archived those as there is useful reference material but we stopped updating.

While non comprehensive, the set of TF tests we run is here (one variant per file): https://github.com/google/iree/tree/main/integrations/tensorflow/test/iree_tf_tests

Those that are not passing are annotated in the file with `XFAIL:`. Each file has a command line to run for the test, and the readme in the containing Tensorflow directory has instructions for running.

While we would love patches on anything that is incomplete for TF, we are generally prioritizing pytorch, TFLite, and Jax compatibility right now and have no active work planned for TF/keras. But if you are something that needs to be fixed, help is appreciated!
